modal_proto/api.proto

```diff
@@ -369,6 +369,9 @@ message AppLookupObjectRequest {
   string object_id = 5;
   string object_entity = 6;
   string environment_name = 7;
+
+  bool lookup_published = 8; // Lookup object on app published by another workspace
```

Comment:
The reason we use a bool instead of breaking into a new `NAMESPACE` is that currently we still wish apps to be deployed into a single namespace i.e. `DEPLOYMENT_NAMESPACE_WORKSPACE`. As such, we cannot simply add a `DEPLOYMENT_NAMESPACE_PUBLISHED`, for example, without weird logic around intra/cross-namespace de-duplication.